File: Following Directions.cpp
#include<bits/stdc++.h>
using namespace std;

int x, y, temp;

int main() {
   int t, b[100];
   char a[100];
   cin >> t;

   for (int i = 0; i < t; i++) {
      int n;
      cin >> n;
      x = 0, y = 0, temp = 0;

      for (int j = 0; j < n; j++) {
         cin >> a[j];
         switch (a[j]) {
            case 'U':
               x += 1;
               break;
            case 'D':
               x -= 1;
               break;
            case 'R':
               y += 1;
               break;
            case 'L':
               y -= 1;
               break;
         }
         if (x == 1 && y == 1) {
            temp = 1;
         }
      }
      if (temp) {
         b[i] = 1;
      } else {
         b[i] = 0;
      }
   }
   for (int i = 0; i < t; i++) {
      if (b[i] == 1) {
         cout << "YES\n";
      } else {
         cout << "NO\n";
      }
   }

   return 0;
}

File: Following Directions.py
t = int(input().strip())
b = []

for _ in range(t):
    n = int(input().strip())
    x, y, temp = 0, 0, 0
    a = input().strip()

    for c in a:
        if c == 'U':
            x += 1
        elif c == 'D':
            x -= 1
        elif c == 'R':
            y += 1
        elif c == 'L':
            y -= 1
        if x == 1 and y == 1:
            temp = 1
            break

    b.append(1 if temp else 0)

for i in range(t):
    print("YES" if b[i] else "NO")
